### ISSUE TITLE
spawner.InstanceSpawner could not be imported

### ISSUE THREAD

**Issue by rubysown** [harvard/cloudJHub#4]
Hi, we are encountering this error when attempting to generate a token via jupyterhub.

```
[C 2017-12-26 23:04:33.828 JupyterHub application:90] Bad config encountered during initialization:
[C 2017-12-26 23:04:33.829 JupyterHub application:91] The 'spawner_class' trait of <jupyterhub.app.JupyterHub object at 0x7f9e45536c88> instance must be a type, but 'spawner.InstanceSpawner' could not be imported
```

Here is our jupyter_config.py:
```
################ Spawner Settings ################
c.JupyterHub.spawner_class = 'spawner.InstanceSpawner'
# The periodicity of checks for user activity. Value is in seconds.
c.JupyterHub.last_activity_interval = 15
# Change this to set the maximum time length between entering passwords
c.JupyterHub.cookie_max_age_days = 1
# Grant admin users permission to access all user notebook servers.
c.JupyterHub.admin_access = True
c.JupyterHub.extra_log_file = '/var/log/jupyterhub'
```

How can we go about importing the Spawner into the config so we can generate a token? Thanks!

**Comment by farassadek** [comment]
Hi, 
I am not sure what cause this issue, but I noticed your config file is  jupyter_config.py instead  jupyterhub_config.py. so If you start jupyterhub from the systemd then file name should be jupyterhub_config.py, otherwise jupyterhub will ignore jupyter_config.py and use jupyterhub default values.

What AMI are you using  ? 
Note that I just installed jupyterhub on ubuntu (latest AMI image ami-aa2ea6d0) and it worked fine.

**Comment by rubysown** [comment]
Hey @farassadek, our config file is named `jupyterhub_config.py`, I just mistyped it in the issue, I apologize.

We are using a custom AMI (with Jupyter stuff pre-loaded) based off of a regular Ubuntu 16.04 AMI.

**Comment by farassadek** [comment]
I guess it is some kind of package version conflicts (like , as an example, older version of traitlets), but I cannot tell for sure unless I run the customized image myself.

**Comment by rubysown** [comment]
@farassadek thanks for your quick responses. Would being on jupyterhub-0.8.2 versus 0.7.2 cause this issue? That is the only version difference I can tell

**Comment by farassadek** [comment]
Yes,
 jupyterhub-8 had some issues therefore we stick with jupyterhub-0.7.2 for now.
please make sure the single-user server and Hub are both running the same version. 

> cd /etc/jupyterhub
> jupyterhub -V
0.7.2
> jupyterhub-singleuser -V
0.7.2

**Comment by rubysown** [comment]
The issue was resolved by downgrading to 0.7.2, we made a slight change to our requirements file which threw us into the dev branch. Thanks for your assistance!
